Storage/AbstractDoctrineStorage.php

```diff
@@ -12,6 +12,13 @@
 abstract class AbstractDoctrineStorage implements StorageInterface
 {
     /**
+     * All the available config storage types.
+     */
+    const STORAGE_ORM     = 'orm';
+    const STORAGE_MONGODB = 'mongodb';
+    const STORAGE_PROPEL  = 'propel';
```

Comment:
You should move the constants into `StorageInterface`. Propel is not a part of Doctrine so it's a bit weird to call `AbstractDoctrineStorage::STORAGE_PROPEL`.